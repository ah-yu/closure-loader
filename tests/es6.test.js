const MemoryFS = require('memory-fs');
const path = require('path');
const compile = require('./helpers/compile');

const config = {
    module: {
        rules: [
            {
                test: /legacy-code\/.*\.js$/,
                use: [
                    {
                        loader: require.resolve('..'),
                        options: {
                            paths: [
                                path.join(
                                    __dirname,
                                    'fixtures',
                                    'es6',
                                    'legacy-code',
                                ),
                            ],
                            es6mode: true,
                            watch: false,
                        },
                    },
                ],
            },
        ],
    },
};

test('loads goog modules with es6 import', async () => {
    const fs = new MemoryFS();
    const stats = (await compile('es6', config, fs)).toJson();
    expect(stats.errors).toHaveLength(0);

    const bundle = fs.readFileSync('/dist/main.bundle.js', 'utf-8');
    const spy = jest.spyOn(console, 'log');

    // eslint-disable-next-line no-eval
    eval(bundle);

    expect(spy).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
});
