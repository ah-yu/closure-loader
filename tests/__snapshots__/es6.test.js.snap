// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads goog modules with es6 import 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "This should contain the empty submodule:",
      Object {
        "value": "This is a module",
      },
    ],
    Array [
      "Call via default export:",
    ],
    Array [
      "this is some.legacy.namespace.myFn",
    ],
    Array [
      "some.legacy.namespace.otherModule.text =",
      "This is from otherModule.",
    ],
    Array [
      "Call via named export:",
    ],
    Array [
      "this is some.legacy.namespace.myFn",
    ],
    Array [
      "some.legacy.namespace.otherModule.text =",
      "This is from otherModule.",
    ],
    Array [
      "Some var:",
      42,
    ],
  ],
}
`;
